comps/llms/rayllm/ray_serve.py

```diff
@@ -0,0 +1,514 @@
+                    )
+                responses.append(
+                    ModelResponse(
+                        generated_text=result[-1],
+                        tool_calls=tool_call_list,
+                        num_input_tokens=self.predictor.input_length,
+                        num_generated_tokens=len(result[-1]),
+                        preprocessing_time=0,
+                    )
+                )
+            return responses
+
+    async def handle_static_batch(self, prompts: List[str], **config: Dict[str, any]):
+        results = self.predictor.generate(prompts, **config)
+        if not self.use_openai:
+            return results
+        else:
+            return ModelResponse(
+                generated_text=results[0].text,
+                num_input_tokens=results[0].input_length,
+                num_input_tokens_batch=results[0].input_length,
+                num_generated_tokens=results[0].generate_length,
+                preprocessing_time=0,
+            )
+
+    def preprocess_prompts(self, input: Union[str, List], tools=None, tool_choice=None):
+        if isinstance(input, str):
+            return input
+        elif isinstance(input, List):
+            prompts = []
+            images = []
+
+            prompt_format = get_prompt_format(input)
+            if prompt_format == PromptFormat.CHAT_FORMAT:
+                # Process the input prompts with tools
+                self.tool_call_list = None
+                self.openai_tools_prompter: OpenAIToolsPrompter = OpenAIToolsPrompter() if tools is not None else None
+                self.tools_capture_texts: ChatPromptCapture = None
+                if self.openai_tools_prompter is not None:
+                    input = self.openai_tools_prompter.inject_prompt(input, tools, tool_choice)
+                    self.tools_capture_texts = ChatPromptCapture()
+                    for m in input:
+                        if m.tool_calls is not None:  # type: ignore
+                            m.content = self.openai_tools_prompter.content_from_assistant(m)  # type: ignore
+                        elif m.tool_call_id is not None:  # type: ignore
+                            m.content = self.openai_tools_prompter.content_from_tool(m)  # type: ignore
+                # Process the input prompts with MLLM tool
+                if self.process_tool is not None:
+                    prompt = self.process_tool.get_prompt(input)
+                    return prompt
+                else:
+                    prompts.extend(input)
+            elif prompt_format == PromptFormat.PROMPTS_FORMAT:
+                prompts.extend(input)
+            else:
+                raise HTTPException(400, "Invalid prompt format.")
+            return prompts
+        else:
+            raise HTTPException(400, "Invalid prompt format.")
+
+    async def __call__(self, http_request: Request) -> Union[StreamingResponse, JSONResponse, str]:
```

Comment:
this function is not used when using openai backend only

Reply:
Thanks for your comments. Have updated it.